### ISSUE TITLE
NEEDS CHANGE:  url = 'http://sick-rage.github.io/sb_network_timezones/network_timezones.txt'

### ISSUE THREAD

**Issue by mmhere** [SickChill/SickChill#5045]
sickbeard/network_timezones.pl

still harbours the old URL using the string sick-rage, which needs to be changed to sickchill.

the system won't restart because it continually tries to get network timezones from the old non-existent location.

I made the change locally in my network_timezones.pl to the following, but am afraid to check it in as I'm not a regular dev here:

http://sickchill.github.io/sb_network_timezones/network_timezones.txt

**Comment by mmhere** [comment]
'scene_loc': 'http://sickchill.github.io/scene_exceptions/scene_exceptions.json',

(in sickbeard/indexers/indexer_config.py) will also solve the scene indexer errors that are otherwise present.  the present/old/incorrect value there is:
'scene_loc': 'http://sick-rage.github.io/scene_exceptions/scene_exceptions.json',
)

**Comment by GldRush98** [comment]
There are also url's _in __init__.py_ and _webserve.py_ that need updated.

**Comment by mmhere** [comment]
Yup.  My little list is by no means complete but I think those are the two that are critical to it starting.

Here are some of my ugly/detailed notes from last week when the string was "sick-rage" -- but note that it is now "sickchill"

--  key URLs change from sickrage.github.io to sick-rage.github.io  --

files modified within here (only two) marked << PREV-20181006--B4-github.io-URL-change >>
    ./sickbeard/webserve.py
    ./sickbeard/__init__.py
    ./sickbeard/network_timezones.py
    ./sickbeard/indexers/indexer_config.py

    # other *.mako --  links/informative only
[N] ./gui/slick/views/config.mako:182:                    <a href="${anon_url('http://sickrage.github.io/')}" rel="noreferrer" onclick="window.open(this.href, '_blank'); return false;">
[N] ./gui/slick/views/config.mako:183:                        http://sickrage.github.io/



    # these aren't strictly necessary:
[Y] ./sickbeard/webserve.py:2395:            changes = helpers.getURL('http://sickrage.github.io/sickrage-news/CHANGES.md', session=helpers.make_session(), returns='text')
[Y] ./sickbeard/webserve.py:2398:            changes = _('Could not load changes from the repo. [Click here for CHANGES.md]({changes_url})').format(changes_url='http://sickrage.github.io/sickrage-news/CHANGES.md')


    # these aren't strictly necessary:
[Y] ./sickbeard/__init__.py:161:NEWS_URL = 'http://sickrage.github.io/sickrage-news/news.md'
[Y] ./sickbeard/__init__.py:162:LOGO_URL = 'http://sickrage.github.io/images/ico/favicon-64.png'
[Y] ./sickbeard/__init__.py:800:            GIT_REMOTE_URL = 'https://github.com/SickRage/SickRage.git'
    --  git related, which we do not operate/interact-with at pure runtime



[Y] ./sickbeard/network_timezones.py:48:    url = 'http://sickrage.github.io/sb_network_timezones/network_timezones.txt'



[Y] ./sickbeard/indexers/indexer_config.py:36:        'scene_loc': 'http://sickrage.github.io/scene_exceptions/scene_exceptions.json',



[N] ./sickbeard/config.py:165:                    "http://sickrage.github.io/unrar/unrar_win.zip", filename=unrar_zip, session=helpers.make_session()
    --  unrar is configured separately on linux, don't care

--> https://github.com/Sick-Rage/Sick-Rage/issues/4984 <--
    new echel0n games

--> https://torrentfreak.com/ongoing-sickrage-dispute-descends-into-trademark-war-181008/ <--
    other media has picked up on it

**Comment by XxUnkn0wnxX** [comment]
there are about 295 files that contain sick-rage.github.io.

for easy use just use code editor guys like BBedit on Mac, or NP++ on windows that is capable of searching for strings of code through all files within a folder.

for BBEdit (Mac Users) do multi-file search, click other and select the root Dir of ur SickChill install then use sick-rage.github.io as the search string & sickchill.github.io as the replace string & replace all. this should replace any instances of sick-rage.github.io through ur entire install.

**Comment by mmhere** [comment]
There are many files that contain problematic URLs but most are comments;  the only files I found critical(*) are `./sickbeard/network_timezones.py` and `./sickbeard/indexers/indexer_config.py`.  In both cases the domain-name part of each URL changes from:
- https://sick-rage.github.io/
  (note that this used to be sickrage.github.io)

to

- https://sickchill.github.io/


The devs have changed most of the URLs already but those two do remain in the current main top-of-trunk.  For remaining ones I'm sure the devs have their own favourite automated editing mechanisms ;-)

(*)I changed those above locally in my particular git clone directory until the devs can get around to it.  I did check in and issue pull requests for these changes but I'm not sure if those will be picked up or if I even did the pull requests properly.  [some codecov complaint popped up on one of them so I probly altered the git pull request text inappropriately.]

As I never git update automatically from the GUI I worried less about git repository links.  I also worried less about news/info links.

I only made the minimal changes to allow the software to start up.  Without a valid URL in network_timezones.py to access network_timezones.txt, for example, it spins its wheels writing endless network URL failures to the log and never launches.  (In the end this merely populates the runtime ./cache.db with network:timezone pairs so this relatively static data *could* be duplicated elsewhere -- see ideas below.)

---

In my humble opinion it could be better to populate the network_timezones.txt data into the respository proper (at least as a backup measure) rather than hosting it as a live file that must be retrieved from github at every startup.  When github has gone away previously (as in the DMCA shenanigans earlier this year from our favourite "friend") or when the github.io website URL has changed (as it has done twice this week), then the local system refuses to start up at all because of the dependency on this live file `network_timezones.txt`.

Just my $0.03, allowing for inflation.

Or how about this:

I realize that it is nice to have the timezone data be live so that it can be changed from the server side w/o having to do a local repo update/clone, but perhaps having both the live network file **and** a fallback static file in the repo would suffice?  I.e., try for the network;  if that fails, load up the local copy.  Refusing to start at all renders the system inoperable, which is what happens now.

Timezones don't change and the entities (transmitter networks) within them change infrequently, so reducing this server dependency for something arguably doesn't *really* requiring being live data, and that has proven to fail a few times this year would be quite nice I think.
